be/src/runtime/exec_env.cpp

```diff
@@ -306,6 +306,23 @@ Status ExecEnv::init(const std::vector<StorePath>& store_paths, bool as_cn) {
             new PriorityThreadPool("table_scan_io", // olap/external table scan thread pool
                                    config::scanner_thread_pool_thread_num, config::scanner_thread_pool_queue_size);
 
+    // Thread pool used for streaming load to scan StreamLoadPipe. The maximum number of
+    // threads and queue size are set INT32_MAX which indicate there is no limit for the
+    // thread pool, and this can avoid deadlock for concurrent streaming loads. The thread
+    // pool will not be full easily because fragment execution pool and http workers also
+    // limit the streaming load concurrency which is controlled by fragment_pool_thread_num_max
+    // and webserver_num_workers respectively. This pool will be used when
+    // enable_streaming_load_thread_pool is true.
+    std::unique_ptr<ThreadPool> streaming_load_pool;
+    RETURN_IF_ERROR(
+            ThreadPoolBuilder("stream_load_io")
+                    .set_min_threads(config::streaming_load_thread_pool_num_min)
+                    .set_max_threads(INT32_MAX)
```

Comment:
this sound like crazy without upper bound limit.

Reply:
Currently we want to solve this problem quickly, so choose the way like StarRocks 2.3 and before. In StarRocks 2.3, there is no limit for the threads used by scanner. But as the comment says, this pool will not be full easily.
```
The maximum number of  threads and queue size are set INT32_MAX which indicate there is no limit for the thread 

pool, and this can avoid deadlock for concurrent streaming loads. The thread pool will not be full easily because 

fragment execution pool and http workers also limit the streaming load concurrency which is controlled by 

`fragment_pool_thread_num_max` and `webserver_num_workers` respectively.
```

Reply:
should be only considered as a temp solution. or set to a number that is not likely to reach, e.g. 32768? in case it reaches, still the thread numbers can be limited, rather than killing the OS.